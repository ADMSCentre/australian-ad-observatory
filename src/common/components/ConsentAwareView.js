import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import useConsent from 'common/hooks/useConsent';
import proceed from 'common/utils/proceed';

const ConsentAwareView = ({ children, whenGrantedIs, whenGrantedIsNot }) => {
  const { granted } = useConsent();

  return proceed(granted, whenGrantedIs, whenGrantedIsNot) ? <Fragment>{children}</Fragment> : null;
};
ConsentAwareView.displayName = 'Consent aware view';

ConsentAwareView.propTypes = {
  children: PropTypes.node.isRequired,
  whenGrantedIs: PropTypes.oneOf(['none', 'old', 'current']),
  whenGrantedIsNot: PropTypes.oneOf(['none', 'old', 'current'])
};

ConsentAwareView.defaultProps = {
  whenGrantedIs: null,
  whenGrantedIsNot: null
};

export default ConsentAwareView;
