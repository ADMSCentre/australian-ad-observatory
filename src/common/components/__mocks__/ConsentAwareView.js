/* eslint-disable react/prop-types */
import React, { Fragment } from 'react';
import proceed from 'common/utils/proceed';

let granted;

export const setGranted = gtd => {
  granted = gtd;
};

const ConsentAwareView = ({ children, whenGrantedIs, whenGrantedIsNot }) =>
  proceed(granted, whenGrantedIs, whenGrantedIsNot) ? <Fragment>{children}</Fragment> : null;

export default ConsentAwareView;
