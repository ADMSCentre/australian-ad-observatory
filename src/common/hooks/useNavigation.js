import { useCallback } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import actions from 'common/state/navigation/actions';

const useNavigation = () => {
  const current = useSelector(({ navigation: { current } }) => current);

  const dispatch = useDispatch();

  const goto = useCallback(destination => dispatch(actions.navigate(destination)), [dispatch]);

  return { current, goto };
};

export default useNavigation;
