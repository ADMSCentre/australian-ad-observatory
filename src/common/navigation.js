import {
  // faGlobe,
  // faShareAlt,
  faInfoCircle,
  faArchive,
  faCog
} from '@fortawesome/free-solid-svg-icons';

export const ITEMS = {
  archive: { icon: faArchive, label: 'My Archive', badge: 0 },
  preferences: { icon: faCog },
  about: { icon: faInfoCircle },
  // news: { icon: faGlobe, badge: 0 },
  // share: { icon: faShareAlt },
  privacy: { label: 'Privacy Policy' },
  terms: { label: 'Terms & Conditions' }
  // help: {}
};
