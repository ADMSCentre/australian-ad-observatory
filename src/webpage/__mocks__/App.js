import React from 'react';

const App = () => <div name="app" />;

export default App;
