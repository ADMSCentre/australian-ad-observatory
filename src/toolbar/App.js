import React, { Fragment } from 'react';
import Container from 'react-bootstrap/Container';

import ConsentAwareView from 'common/components/ConsentAwareView';
import Button from 'common/components/Button';
import MainMenu from './components/MainMenu';

import 'bootstrap/dist/css/bootstrap.min.css';
import './App.less';

const App = () => (
  <Fragment>
    <ConsentAwareView whenGrantedIs="none">
      <Container>
        <h3>{process.env.TITLE}</h3>
        <div className="mb-1">
          <p>To enable the extension, please consent to our collection of your data.</p>
          <Button label="Review Consent" to="terms" />
        </div>
      </Container>
    </ConsentAwareView>
    <ConsentAwareView whenGrantedIs="old">
      <Container>
        <div className="mb-1">
          <h3>{process.env.TITLE}</h3>
          <p>
            To enable the extension, please consent to our collection of some more of your data.
          </p>
          <Button label="Review Consent" to="terms" />
        </div>
      </Container>
    </ConsentAwareView>
    <ConsentAwareView whenGrantedIs="current">
      <MainMenu />
    </ConsentAwareView>
  </Fragment>
);
App.displayName = 'App';

export default App;
