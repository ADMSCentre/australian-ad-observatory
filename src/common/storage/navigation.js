import { CURRENT_LOCATION } from 'common/keys';
import { getItem, setItem } from '.';

/**
 *  get the current location (a tab name)
 *
 *  @return the current saved location
 */
export const loadCurrent = async () => getItem(CURRENT_LOCATION);

/**
 *  Saves the current location (a tab name)
 *
 *  @param current â€” A tab name
 */
export const saveCurrent = async current => setItem({ [CURRENT_LOCATION]: current });
