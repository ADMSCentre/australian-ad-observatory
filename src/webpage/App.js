import React, { Fragment } from 'react';
import Container from 'react-bootstrap/Container';

// import useArchive from 'common/hooks/useArchive';
import useNavigation from 'common/hooks/useNavigation';

import ConsentAwareView from 'common/components/ConsentAwareView';

import Header from './components/Header';
import Navigation from './components/Navigation';

// import NewsPage from './pages/NewsPage';
import AboutPage from './pages/AboutPage';
import ArchivePage from './pages/ArchivePage';
import PreferencesPage from './pages/PreferencesPage';
// import SharePage from './pages/SharePage';
import PrivacyPage from './pages/PrivacyPage';
import TermsPage from './pages/TermsPage';
import HelpPage from './pages/HelpPage';

import 'bootstrap/dist/css/bootstrap.min.css';
import './App.less';

const routes = {
  // news: NewsPage,
  about: AboutPage,
  archive: ArchivePage,
  preferences: PreferencesPage,
  // share: SharePage,
  privacy: PrivacyPage,
  terms: TermsPage,
  help: HelpPage
};

const App = () => {
  const { current } = useNavigation();
  const Page = routes[current] || AboutPage;

  return (
    <Fragment>
      <Header />
      <div className="d-flex" id="wrapper">
        <ConsentAwareView whenGrantedIsNot="current">
          <div id="page-content-wrapper">
            <TermsPage />
          </div>
        </ConsentAwareView>
        <ConsentAwareView whenGrantedIs="current">
          <Navigation />
          <div id="page-content-wrapper">
            <Container fluid>
              <Page />
            </Container>
          </div>
        </ConsentAwareView>
      </div>
    </Fragment>
  );
};
App.displayName = 'App';

export default App;
