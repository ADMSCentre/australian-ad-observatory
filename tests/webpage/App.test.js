import React from 'react';
import renderer from 'react-test-renderer';
import App from 'webpage/App';

import useNavigation from 'common/hooks/useNavigation';
import { setGranted } from 'common/components/ConsentAwareView';
import { ITEMS } from 'common/navigation';

jest.mock('common/hooks/useNavigation');
// jest.mock('webpage/pages/NewsPage');
jest.mock('webpage/pages/AboutPage');
jest.mock('webpage/pages/ArchivePage');
jest.mock('webpage/pages/PreferencesPage');
// jest.mock('webpage/pages/SharePage');
// jest.mock('webpage/pages/PrivacyPage');
jest.mock('webpage/pages/TermsPage');
jest.mock('webpage/pages/HelpPage');
jest.mock('webpage/components/Navigation');
jest.mock('common/components/ConsentAwareView');

[...Object.keys(ITEMS), ''].forEach(tab => {
  describe(`when tab is '${tab}'`, () => {
    let tree;

    describe('when granted is "current"', () => {
      beforeAll(() => {
        setGranted('current');
        useNavigation.mockReturnValue({ current: tab });
        tree = renderer.create(<App />);
      });

      it(`renders the ${tab} page with correct navigation`, () => {
        expect(tree).toMatchSnapshot();
      });
    });

    describe('when granted is not "current"', () => {
      beforeAll(() => {
        setGranted('none');
        useNavigation.mockReturnValue({ current: tab });
        tree = renderer.create(<App />);
      });

      it('renders the terms page with no navigation', () => {
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
