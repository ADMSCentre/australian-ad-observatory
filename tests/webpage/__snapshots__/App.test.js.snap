// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when tab is '' when granted is "current" renders the  page with correct navigation 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      <img
        className="d-inline-block align-top"
        height={30}
        src="test-file-stub"
        width={30}
      />
      <span
        className="pl-2"
      >
        <strong>
          Test Project
        </strong>
      </span>
    </a>
  </nav>,
  <div
    className="d-flex"
    id="wrapper"
  >
    <div
      id="navigation"
    >
      Navigation goes here
    </div>
    <div
      id="page-content-wrapper"
    >
      <div
        className="container-fluid"
      >
        <h1>
          About Page goes here
        </h1>
      </div>
    </div>
  </div>,
]
`;

exports[`when tab is '' when granted is not "current" renders the terms page with no navigation 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      <img
        className="d-inline-block align-top"
        height={30}
        src="test-file-stub"
        width={30}
      />
      <span
        className="pl-2"
      >
        <strong>
          Test Project
        </strong>
      </span>
    </a>
  </nav>,
  <div
    className="d-flex"
    id="wrapper"
  >
    <div
      id="page-content-wrapper"
    >
      <h1>
        Terms Page goes here
      </h1>
    </div>
  </div>,
]
`;

exports[`when tab is 'about' when granted is "current" renders the about page with correct navigation 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      <img
        className="d-inline-block align-top"
        height={30}
        src="test-file-stub"
        width={30}
      />
      <span
        className="pl-2"
      >
        <strong>
          Test Project
        </strong>
      </span>
    </a>
  </nav>,
  <div
    className="d-flex"
    id="wrapper"
  >
    <div
      id="navigation"
    >
      Navigation goes here
    </div>
    <div
      id="page-content-wrapper"
    >
      <div
        className="container-fluid"
      >
        <h1>
          About Page goes here
        </h1>
      </div>
    </div>
  </div>,
]
`;

exports[`when tab is 'about' when granted is not "current" renders the terms page with no navigation 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      <img
        className="d-inline-block align-top"
        height={30}
        src="test-file-stub"
        width={30}
      />
      <span
        className="pl-2"
      >
        <strong>
          Test Project
        </strong>
      </span>
    </a>
  </nav>,
  <div
    className="d-flex"
    id="wrapper"
  >
    <div
      id="page-content-wrapper"
    >
      <h1>
        Terms Page goes here
      </h1>
    </div>
  </div>,
]
`;

exports[`when tab is 'archive' when granted is "current" renders the archive page with correct navigation 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      <img
        className="d-inline-block align-top"
        height={30}
        src="test-file-stub"
        width={30}
      />
      <span
        className="pl-2"
      >
        <strong>
          Test Project
        </strong>
      </span>
    </a>
  </nav>,
  <div
    className="d-flex"
    id="wrapper"
  >
    <div
      id="navigation"
    >
      Navigation goes here
    </div>
    <div
      id="page-content-wrapper"
    >
      <div
        className="container-fluid"
      >
        <h1>
          Archive Page goes here
        </h1>
      </div>
    </div>
  </div>,
]
`;

exports[`when tab is 'archive' when granted is not "current" renders the terms page with no navigation 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      <img
        className="d-inline-block align-top"
        height={30}
        src="test-file-stub"
        width={30}
      />
      <span
        className="pl-2"
      >
        <strong>
          Test Project
        </strong>
      </span>
    </a>
  </nav>,
  <div
    className="d-flex"
    id="wrapper"
  >
    <div
      id="page-content-wrapper"
    >
      <h1>
        Terms Page goes here
      </h1>
    </div>
  </div>,
]
`;

exports[`when tab is 'preferences' when granted is "current" renders the preferences page with correct navigation 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      <img
        className="d-inline-block align-top"
        height={30}
        src="test-file-stub"
        width={30}
      />
      <span
        className="pl-2"
      >
        <strong>
          Test Project
        </strong>
      </span>
    </a>
  </nav>,
  <div
    className="d-flex"
    id="wrapper"
  >
    <div
      id="navigation"
    >
      Navigation goes here
    </div>
    <div
      id="page-content-wrapper"
    >
      <div
        className="container-fluid"
      >
        <h1>
          Preferences Page goes here
        </h1>
      </div>
    </div>
  </div>,
]
`;

exports[`when tab is 'preferences' when granted is not "current" renders the terms page with no navigation 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      <img
        className="d-inline-block align-top"
        height={30}
        src="test-file-stub"
        width={30}
      />
      <span
        className="pl-2"
      >
        <strong>
          Test Project
        </strong>
      </span>
    </a>
  </nav>,
  <div
    className="d-flex"
    id="wrapper"
  >
    <div
      id="page-content-wrapper"
    >
      <h1>
        Terms Page goes here
      </h1>
    </div>
  </div>,
]
`;

exports[`when tab is 'privacy' when granted is "current" renders the privacy page with correct navigation 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      <img
        className="d-inline-block align-top"
        height={30}
        src="test-file-stub"
        width={30}
      />
      <span
        className="pl-2"
      >
        <strong>
          Test Project
        </strong>
      </span>
    </a>
  </nav>,
  <div
    className="d-flex"
    id="wrapper"
  >
    <div
      id="navigation"
    >
      Navigation goes here
    </div>
    <div
      id="page-content-wrapper"
    >
      <div
        className="container-fluid"
      >
        <header>
          <h1>
            Privacy
          </h1>
        </header>
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>Privacy!</p>",
            }
          }
        />
      </div>
    </div>
  </div>,
]
`;

exports[`when tab is 'privacy' when granted is not "current" renders the terms page with no navigation 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      <img
        className="d-inline-block align-top"
        height={30}
        src="test-file-stub"
        width={30}
      />
      <span
        className="pl-2"
      >
        <strong>
          Test Project
        </strong>
      </span>
    </a>
  </nav>,
  <div
    className="d-flex"
    id="wrapper"
  >
    <div
      id="page-content-wrapper"
    >
      <h1>
        Terms Page goes here
      </h1>
    </div>
  </div>,
]
`;

exports[`when tab is 'terms' when granted is "current" renders the terms page with correct navigation 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      <img
        className="d-inline-block align-top"
        height={30}
        src="test-file-stub"
        width={30}
      />
      <span
        className="pl-2"
      >
        <strong>
          Test Project
        </strong>
      </span>
    </a>
  </nav>,
  <div
    className="d-flex"
    id="wrapper"
  >
    <div
      id="navigation"
    >
      Navigation goes here
    </div>
    <div
      id="page-content-wrapper"
    >
      <div
        className="container-fluid"
      >
        <h1>
          Terms Page goes here
        </h1>
      </div>
    </div>
  </div>,
]
`;

exports[`when tab is 'terms' when granted is not "current" renders the terms page with no navigation 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      <img
        className="d-inline-block align-top"
        height={30}
        src="test-file-stub"
        width={30}
      />
      <span
        className="pl-2"
      >
        <strong>
          Test Project
        </strong>
      </span>
    </a>
  </nav>,
  <div
    className="d-flex"
    id="wrapper"
  >
    <div
      id="page-content-wrapper"
    >
      <h1>
        Terms Page goes here
      </h1>
    </div>
  </div>,
]
`;
